integration_test.go

```diff
@@ -560,3 +564,707 @@ func TestNewConnectWithLowTimeout(t *testing.T) {
+type SliceMapTypesTestCase struct {
+	CQLType           string
+	CQLValue          string      // Non-NULL value to insert
+	ExpectedValue     interface{} // Expected value for non-NULL case
+	ExpectedNullValue interface{} // Expected value for NULL
+}
+
+// compareCollectionValues compares collection values (lists, sets, maps) with special handling
+func compareCollectionValues(t *testing.T, cqlType string, expected, actual interface{}) bool {
+	switch {
+	case strings.HasPrefix(cqlType, "set<"):
+		// Sets are returned as slices, but order is not guaranteed
+		expectedSlice := reflect.ValueOf(expected)
+		actualSlice := reflect.ValueOf(actual)
+		if expectedSlice.Kind() != reflect.Slice || actualSlice.Kind() != reflect.Slice {
+			return false
+		}
+		if expectedSlice.Len() != actualSlice.Len() {
+			return false
+		}
+
+		// Convert to maps for unordered comparison
+		expectedSet := make(map[interface{}]bool)
+		for i := 0; i < expectedSlice.Len(); i++ {
+			expectedSet[expectedSlice.Index(i).Interface()] = true
+		}
+
+		actualSet := make(map[interface{}]bool)
+		for i := 0; i < actualSlice.Len(); i++ {
+			actualSet[actualSlice.Index(i).Interface()] = true
+		}
+
+		return reflect.DeepEqual(expectedSet, actualSet)
+
+	default:
+		// For lists, maps, and other collections, reflect.DeepEqual works fine
+		return reflect.DeepEqual(expected, actual)
+	}
+}
+
+// compareValues compares expected and actual values with type-specific logic
+func compareValues(t *testing.T, cqlType string, expected, actual interface{}) bool {
+	switch cqlType {
+	case "varint":
+		// big.Int needs Cmp() for proper comparison, but handle nil pointers safely
+		if expectedBig, ok := expected.(*big.Int); ok {
+			if actualBig, ok := actual.(*big.Int); ok {
+				// Handle nil cases
+				if expectedBig == nil && actualBig == nil {
+					return true
+				}
+				if expectedBig == nil || actualBig == nil {
+					return false
+				}
+				return expectedBig.Cmp(actualBig) == 0
+			}
+		}
+		return reflect.DeepEqual(expected, actual)
+
+	case "decimal":
+		// inf.Dec needs Cmp() for proper comparison, but handle nil pointers safely
+		if expectedDec, ok := expected.(*inf.Dec); ok {
+			if actualDec, ok := actual.(*inf.Dec); ok {
+				// Handle nil cases
+				if expectedDec == nil && actualDec == nil {
+					return true
+				}
+				if expectedDec == nil || actualDec == nil {
+					return false
+				}
+				return expectedDec.Cmp(actualDec) == 0
+			}
+		}
+		return reflect.DeepEqual(expected, actual)
+
+	default:
+		// reflect.DeepEqual handles nil vs empty slice/map distinction correctly for all types
+		// including inet (net.IP), blob ([]byte), collections ([]T, map[K]V), etc.
+		// This is critical for catching zero value behavior changes in the driver
+		return reflect.DeepEqual(expected, actual)
+	}
+}
+
+// TestSliceMapMapScanTypes tests SliceMap and MapScan with various CQL types
+func TestSliceMapMapScanTypes(t *testing.T) {
```

Comment:
Could you please move all serialization tests to `integration_serialization_scylla_test.go`